src/NuGet.CommandLine.StagingExtensions/StageManagementResource.cs

```diff
@@ -107,6 +109,39 @@ public StageManagementResource(string stageServiceUri, HttpSource httpSource)
             return result;
         }
 
+        public async Task<IReadOnlyList<StageListView>> List(string apiKey, Common.ILogger log)
+        {
+            if (string.IsNullOrWhiteSpace(apiKey))
+            {
+                throw new ArgumentException(StagingResources.ApiKeyShouldNotBeEmpty, nameof(apiKey));
+            }
+
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            var result = await _httpSource.ProcessResponseAsync(
+                () =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, new Uri(new Uri(_stageServiceUri), $"{StagePath}"));

```

Comment:
It looks liek _stageServiceUri is saved away as a string, but then turned into a Uri in every case it is used in this class. Is there a reason that this private variable isn't already saved away as a Uri? Or can we change it to that wihtout issue?

---
In reply to: [74158586](https://github.com/NuGet/NuGet.Staging/pull/68#discussion_r74158586) [](ancestors = 74158586)